cpp/include/raft/core/resource/nccl_clique.hpp

```diff
@@ -25,38 +25,102 @@ namespace raft::resource {
 
 class nccl_clique_resource : public resource {
  public:
-  nccl_clique_resource() : clique_(std::make_unique<raft::comms::nccl_clique>()) {}
+  nccl_clique_resource(std::optional<std::vector<int>>& device_ids, int percent_of_free_memory)
+  {
+    if (device_ids.has_value()) {
+      clique_ = std::make_unique<raft::core::nccl_clique>(*device_ids, percent_of_free_memory);
+    } else {
+      clique_ = std::make_unique<raft::core::nccl_clique>(percent_of_free_memory);
```

Comment:
What happens when the percent of free memory is used up? Does it automatically try to allocate more? Please document this class and the constructor.